lib/hashes.md

```diff
@@ -5,40 +5,97 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Anything where you have to associate 2 ruby objects: A name and an email address, a menu item and a price, a novel
+and an author, etc. 
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Hashes and Arrays are similar in that they are both collections of variables. Neither an Array nor a Hash has to be all numbers,
+or all strings, or all floats, etc., all can be combinations of different types of variables. They are different in that Arrays
+have an enumerated order to them and Hashes do not. To go back to the description of a key-value pair I used in the "What is a Hash" question
+with an Array the name of the container (the key) would always be a number, and the value could be a variety of things. With a Hash
+the name of the container could be any Ruby object.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+Retrieve a particular value by using its key. After many failures I did this with a simple hash in repl.it.
+#NOTE I am including all of these bits of code that I try in repl.it because I am saving these as notes pages, and
+I want to include the code in my notes pages. I am not trying to see how much code I can turn in or anything like that.
+I know you aren't looking for code.
+
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines"}
+puts StateCaps2.keys.inspect
+puts StateCaps2.values.inspect
+puts StateCaps2["Vermont"] #This returns just the capital of Vermont, which is "Montpelier.
+
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+There are probably several ways to do this, but I found a Merge method that works kind of like "join".
+I made a second hash of more states and capitals, then used the merge! method to make one larger hash.
+This tacked on the new variables at visually at least is the "end" of the old hash.
+StateCaps2 = Hash.new{}
+StateCaps2 = {"Idaho"=>"Boise", "Oregon"=>"Salem", "Vermont"=>"Montpelier", "Iowa"=>"Des Moines"}
+StateCaps3 = {"New York"=>"Albany", "Texas"=>"Austin", "Kentucky"=>"Frankfort"}
+StateCaps2.merge! StateCaps3
+puts StateCaps2
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+This seems like something that calls for a loop, specifically an "each do end" loop.
+The bit of code below is an adaptation of found material, which isn't my favorite way to do this, 
+and I'm still working backwards a little bit to say what everything does. Mostly I am taking out
+pieces that I don't understand and seeing what error message I get. I get the impression that 
+the "#" before {key} and {value} tells Ruby to evaluate each key and value in the hash and loop them into the appropriate places
```

Comment:
That's correct. It might help to consider the following Ruby code:

```ruby
name = "Sumeet"

# The following two lines are 100% equal, in that they do the same exact thing:
puts "My name is #{name}"
puts "My name is " + name
```

Both lines result in the text `"My name is Sumeet"` being displayed. The first approach is called "string interpolation", and the second is called "string concatenation". In Ruby, we tend to prefer interpolation over concatenation.